### DISCUSSION TITLE
Updating to 4.15.0 - cause an error when generating excel reports

### DISCUSSION THREAD

**Discussion by tameergh1** [eclipse-birt/birt#1716] [General]
I'm using an old version = 4.8.0.
After updating to 4.15.0 all the formats work as expected except excel format, I'm getting this error:
" org.eclipse.birt.core.exception.BirtException: Unable to save file (\"{}\")
at uk.co.spudsoft.birt.emitters.excel.ExcelEmitter.end(ExcelEmitter.java:336)
at org.eclipse.birt.report.engine.api.impl.RenderTask$PageRangeRender.render(RenderTask.java:585)
at org.eclipse.birt.report.engine.api.impl.RenderTask.render(RenderTask.java:278)
... 6 more
Caused by: java.lang.ExceptionInInitializerError
at org.apache.poi.openxml4j.opc.internal.ContentTypeManager.save(ContentTypeManager.java:429)
at org.apache.poi.openxml4j.opc.ZipPackage.saveImpl(ZipPackage.java:608)
at org.apache.poi.openxml4j.opc.OPCPackage.save(OPCPackage.java:1532)
at org.apache.poi.ooxml.POIXMLDocument.write(POIXMLDocument.java:227)
at uk.co.spudsoft.birt.emitters.excel.ExcelEmitter.end(ExcelEmitter.java:331)
... 8 more
Caused by: java.lang.ClassCastException: class org.apache.xerces.parsers.XIncludeAwareParserConfiguration cannot be cast to class org.apache.xerces.xni.parser.XMLParserConfiguration (org.apache.xerces.parsers.XIncludeAwareParserConfiguration is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader; org.apache.xerces.xni.parser.XMLParserConfiguration is in unnamed module of loader 'app')
at org.apache.xerces.parsers.DOMParser.<init>(Unknown Source)
at org.apache.xerces.parsers.DOMParser.<init>(Unknown Source)
at org.apache.xerces.jaxp.DocumentBuilderImpl.<init>(Unknown Source)at org.apache.xerces.jaxp.DocumentBuilderFactoryImpl.newDocumentBuilder(Unknown Source)
at org.apache.poi.util.XMLHelper.newDocumentBuilder(XMLHelper.java:141)
at org.apache.poi.ooxml.util.DocumentHelper.newDocumentBuilder(DocumentHelper.java:47)
at org.apache.poi.ooxml.util.DocumentHelper.<clinit>(DocumentHelper.java:36)
... 13 more\n]

Caused by: java.lang.ClassCastException: class org.apache.xerces.parsers.XIncludeAwareParserConfiguration cannot be cast to class org.apache.xerces.xni.parser.XMLParserConfiguration (org.apache.xerces.parsers.XIncludeAwareParserConfiguration is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader ; org.apache.xerces.xni.parser.XMLParserConfiguration is in unnamed module of loader 'app')

at org.apache.xerces.parsers.DOMParser.<init>(Unknown Source)"

What could cause this error ??

**Comment by speckyspooky** [comment]
Hello @tameergh1,
the first question which version you have used based on the BIRT-download site the version 4.15 or via eclipse download?

I took a look at your StackTrace and figured out the important information here:
` at uk.co.spudsoft.birt.emitters.excel.ExcelEmitter.end(ExcelEmitter.java:331)`

There the excel-emitter check the usage of excel-templates which seems to be your first error (it is on crash).
And the second one is that the save-operation of the excel file is not possible.
Can ypu check if your report use an excel-template, remove it and try it again.
In addition on it if you have tested the new user property "ExcelEmitter.STREAMING_XLSX" for streaming files, please turn it off if it is used.
This is an additioanl option in the near of the streaming-configuration of "STREAMING_XLSX".
`at uk.co.spudsoft.birt.emitters.excel.ExcelEmitter.end(ExcelEmitter.java:336)`

I have tested lot of excel export with version 4.15 which works well without you exception.

If you have a simple example report without external resources and data access I would be able to debug your report.

**Reply by tameergh1** [reply]
I downloaded birt-runtime-osgi from here https://download.eclipse.org/birt/updates/release/4.15.0/index.html
I'm not using this new user prop
what do you mean by using excel-template? actually the same template is used for all  the report formats

**Reply by speckyspooky** [reply]
Ok, I wor with the same download-site.

I mean that the error reference internally of BIRT on to possible user properties which can be added on report-level
to control the excel behavior and this is in your case "ExcelEmitter.TemplateFile" & "ExcelEmitter.StreamingXlsx".

If you have not set both user properties then it seems a special topic with your report.
Then we would need an simple example of your report with this error to reproduce it at a debug session.

**Reply by tameergh1** [reply]
I tried a simple example that is attached in one of the conversation here that should work and it doesn't

**Reply by merks** [reply]
Could you link to that conversation with the attached test case?

**Reply by tameergh1** [reply]
Sure
https://github.com/eclipse-birt/birt/discussions/1688#discussion-6676144

**Reply by speckyspooky** [reply]
Have you seen my comments there 2 weeks ago?
I have retested there mentioned report and all was fine. I will retest again your report with 4.15 and latest 4.16.

Do you use Windows or Lunux or Mac?
My tests was done with the All-in-One-Designer for Windows.

**Reply by speckyspooky** [reply]
I have done the retests with the folowing result:

- used report #1688: WhiteSpaceTestReport.rptdesign

**Test cases BIRT 4.15**
- case 01: BIRT 4.15, Windows, spudsoft-excel XLSX from HTML-viewer, excel result: ok
- case 02: BIRT 4.15, Windows, spudsoft-excel XLSX from designer, excel result: ok
- case 03: BIRT 4.15, Windows, spudsoft-excel XLS from HTML-viewer, excel result: ok
- case 04: BIRT 4.15, Windows, spudsoft-excel XLS from designer, excel result: ok

**Test cases BIRT 4.16**
- case 05: BIRT 4.16, Windows, spudsoft-excel XLSX from HTML-viewer, excel result: ok
- case 06: BIRT 4.16, Windows, spudsoft-excel XLSX from designer, excel result: ok
- case 07: BIRT 4.16, Windows, spudsoft-excel XLS from HTML-viewer, excel result: ok
- case 08: BIRT 4.16, Windows, spudsoft-excel XLS from designer, excel result: ok

Currently I'm not able to recreate the error with this demo report for BIRT under Windows.

### Screen BIRT 4.15 (Release, Windows), report: WhiteSpaceTestReport.rptdesign
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/92ae6dd1-6dab-4c7f-8a5f-5a5df617113d)

### Screen BIRT 4.16 (Dev-version, Windows), report: WhiteSpaceTestReport.rptdesign
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/99ca2241-8e8a-4ce5-b697-6672eb051850)

**Reply by tameergh1** [reply]
I used the all in one for windows,
The question is why it is work for other formats such as pdf,html,ppt...

And if 4.8 works with additional changes to make excel works.
And now, when trying to use the existing excel emitter maybe I have a mismatch configuration or loaded a wrong class for xml parsering?!

**Reply by speckyspooky** [reply]
The emitter execution between the outputs is for every output different so we cannot come the working for pdf, html and ppt with excel directly. Your StackTrace can only appears with the excel output which is produced from the excel emitter.

Have you another test report with the same problem or only the mentioned test case?
And I think you haven't changed classes or jar-files from the standard installation (please correct me if you have changed something).

**Reply by tameergh1** [reply]
@speckyspooky 
In order to be sure that I'm doing it correctly, when I want to use a new desinger, I need just to change the rptdesign version right?

**Reply by speckyspooky** [reply]
I opend the report with the designer without any changes and started directly the execution.
I got no hint that the version will be changed. And I changed the version number not manually.

So I was able to start the report really directly.

**Comment by tameergh1** [comment]
No, I did not
Ok, what could cause the excel emitter to fall with this error?

**Reply by speckyspooky** [reply]
It coul be that something is special with the report which cause the issue on your side.

I have added another test-report based on the "classic cars data". Could you please try this report with your BIRT 4.15.
The zip-file contains also the excel-xlsx-result from my BIRT-installation. The report is created with the released version of BIRT 4.15.

### Demo report (new), created with BIRT 4.15 202403270652
[birt_415_excel.zip](https://github.com/user-attachments/files/15526437/birt_415_excel.zip)

### Screen of the output result
![grafik](https://github.com/eclipse-birt/birt/assets/41593722/498c702d-2fe9-41bd-abcb-af2d03face8c)
